pkg/clusteragent/clusterchecks/dispatcher_nodes.go

```diff
@@ -122,3 +136,32 @@ func (d *dispatcher) expireNodes() {
 		log.Warn("No nodes reporting, cluster checks will not run")
 	}
 }
+
+// updateRunnersStats collects stats from the registred
+// Cluster Level Check runners and updates the stats cache
+func (d *dispatcher) updateRunnersStats() {
+	if d.clcRunnersClient == nil {
+		log.Debug("Cluster Level Check runner client was not correctly initialised")
+		return
+	}
+
+	d.store.Lock()
+	defer d.store.Unlock()
+	for name, node := range d.store.nodes {
+		node.RLock()
+		ip := node.clientIP
+		node.RUnlock()
+
+		stats, err := d.clcRunnersClient.GetRunnerStats(ip)
+		if err != nil {
+			log.Debugf("Cannot get CLC Runner stats with IP %s on node %s: %v", node.clientIP, name, err)
+		} else {
+			node.Lock()
+			node.clcRunnerStats = stats
+			log.Tracef("Updated CLC Runner stats on node: %s, node IP: %s, stats: %v", name, node.clientIP, stats)
+			node.busyness = calculateBusyness(stats)
+			node.Unlock()
+			log.Debugf("Updated busyness on node: %s, node IP: %s, busyness value: %d", name, node.clientIP, node.busyness)
```

Comment:
Why not include this in the above lock, or have an RLock?

Reply:
yes it should be included in the above lock 👍